simulation/sub8_gazebo/nodes/job_runner.py

```diff
@@ -0,0 +1,264 @@
+                        f.write(response.message + '\n')
+                        f.write("-----------------------------------------------------------")
+                        f.write("\n")
+                except IOError:
+                    print "Diagnostics file not found. Creating it now."
+                    with open(DIAG_OUT_URI + 'diag.txt', 'w') as f:
+                        f.write("Status: {0} occured at {1} .\n".format(response.success, int(time.time())))
+                        f.write(response.message + '\n')
+                        f.write("-----------------------------------------------------------")
+                        f.write("\n")
+                print
+                print "Done!"
+
+            print "------------------------------------------------------------------"
+            print "{0} Successes, {1} Fails, {2} Total".format(self.successes, self.fails, current_loop)
+            print "------------------------------------------------------------------"
+            print
+
+            if current_loop > 5 and self.fails / current_loop > .9:
+                print "Too many errors. Stopping as to not fill HDD with bags."
+                self.loop_count = 0
+            #self.reset_gazebo()
+        print
+        print "Job Finished!"
+        print "Writing report to file. Do not exit."
+        with open(DIAG_OUT_URI + 'diag.txt', 'a') as f:
+            f.write("{0} Successes, {1} Fails, {2} Total \n.".format(self.successes, self.fails, current_loop))
+            f.write("Time of completion: {0}. \n".format(int(time.time())))
+            f.write("-----------------------------------------------------------")
+            f.write("\n")
+
+    def reset_gazebo(self):
+        '''
+        Reset sub position without breaking the simulator.
+        This could send a signal to the mission node so that it can set the sub position wherever it wants.
+        '''
+        print "Reseting Gazebo"
+        return
+
+    @staticmethod
+    @txros.util.cancellableInlineCallbacks
+    def set_model_position(nh, pose, twist=None, model='sub8'):
+        '''
+        Set the position of 'model' to 'pose'.
+        It may be helpful to kill the sub before moving it.
+
+        TODO: Make this a service? Might as well just use the ros service gazebo provides? Maybe not.
+        '''
+        print "Setting position"
+        set_state = nh.get_service_client('/gazebo/set_model_state', SetModelState)
+
+        if twist is None:
+            twist = Twist()
+            twist.linear.x = 0
+            twist.linear.y = 0
+            twist.linear.z = 0
+
+        model_state = ModelState()
+        model_state.model_name = model
+        model_state.pose = pose
+        model_state.twist = twist
+
+        if model == 'sub8':
+            kill = nh.get_service_client('/set_kill', SetKill)
+            yield kill(SetKillRequest(kill=Kill(active=True)))
```

Comment:
Love it

- Jake

> On Apr 30, 2016, at 12:00 AM, mattlangford <notifications@github.com> wrote:
> 
> In simulation/sub8_gazebo/nodes/job_runner.py:
> 
> > +        set_state = nh.get_service_client('/gazebo/set_model_state', SetModelState)
> > +
> > +        if twist is None:
> > +            twist = Twist()
> > +            twist.linear.x = 0
> > +            twist.linear.y = 0
> > +            twist.linear.z = 0
> > +
> > +        model_state = ModelState()
> > +        model_state.model_name = model
> > +        model_state.pose = pose
> > +        model_state.twist = twist
> > +
> > +        if model == 'sub8':
> > +            kill = nh.get_service_client('/set_kill', SetKill)
> > +            yield kill(SetKillRequest(kill=Kill(active=True)))
> This seems like a jank way to reset the sub's position but it worked. Let me know if you think this should be changed.
> 
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
>